src/pages/library/library-provider/library.ts

```diff
@@ -0,0 +1,119 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { AppConfigs } from "../../../providers/appConfig";
+import { ApiProvider } from "../../../providers/api/api";
+import { UtilsProvider } from "../../../providers/utils/utils";
+
+/*
+  Generated class for the LibraryProvider provider.
+
+  See https://angular.io/guide/dependency-injection for more info on providers
+  and Angular DI.
+*/
+@Injectable()
+export class LibraryProvider {
+  constructor(
+    public http: HttpClient,
+    public apiService: ApiProvider,
+    public utils: UtilsProvider
+  ) {
+    console.log("Hello LibraryProvider Provider");
+  }
+
+  getObservationSolutionsList() {
+    this.utils.startLoader();
+    return new Promise((resolve, reject) => {
+      const url = AppConfigs.library.observationSolutionsList;
+
+      this.apiService.httpGet(
+        url,
+        (successData) => {
+          this.utils.stopLoader();
+          resolve(successData.result);
+        },
+        (error) => {
+          this.utils.stopLoader();
+          reject();
+        }
+      );
+    });
+  }
+
+  getSolutiontemplate(solutionId) {
+    this.utils.startLoader();
+    return new Promise((resolve, reject) => {
+      const url = AppConfigs.library.solutionTemplateDetail + solutionId;
+
+      this.apiService.httpGet(
+        url,
+        (successData) => {
+          this.utils.stopLoader();
+          resolve(successData.result);
+        },
+        (error) => {
+          this.utils.stopLoader();
+          reject();
+        }
+      );
+    });
+  }
+
+  getObservationMetaForm(solutionId) {
```

Comment:
rename to getSolutionMetaForm so that ist generic assuming its  same api

Reply:
done